thirdparty/patches/opencensus-cpp-shutdown-api.patch

```diff
@@ -1,7 +1,33 @@
-diff --git opencensus/stats/internal/delta_producer.cc opencensus/stats/internal/delta_producer.cc
-index c61b4d9..b3e4ef2 100644
---- opencensus/stats/internal/delta_producer.cc
-+++ opencensus/stats/internal/delta_producer.cc
+diff --git a/opencensus/exporters/stats/prometheus/internal/prometheus_exporter.cc b/opencensus/exporters/stats/prometheus/internal/prometheus_exporter.cc
+index 3bb5962..706b9b4 100644
+--- a/opencensus/exporters/stats/prometheus/internal/prometheus_exporter.cc
++++ b/opencensus/exporters/stats/prometheus/internal/prometheus_exporter.cc
+@@ -25,7 +25,7 @@ namespace opencensus {
+ namespace exporters {
+ namespace stats {
+ 
+-std::vector<prometheus::MetricFamily> PrometheusExporter::Collect() const {
++std::vector<prometheus::MetricFamily> PrometheusExporter::Collect() {
+   const auto data = opencensus::stats::StatsExporter::GetViewData();
+   std::vector<prometheus::MetricFamily> output(data.size());
+   for (int i = 0; i < data.size(); ++i) {
+diff --git a/opencensus/exporters/stats/prometheus/prometheus_exporter.h b/opencensus/exporters/stats/prometheus/prometheus_exporter.h
+index bbb285d..ab6471f 100644
+--- a/opencensus/exporters/stats/prometheus/prometheus_exporter.h
++++ b/opencensus/exporters/stats/prometheus/prometheus_exporter.h
+@@ -41,7 +41,7 @@ namespace stats {
+ // PrometheusExporter is thread-safe.
+ class PrometheusExporter final : public ::prometheus::Collectable {
+  public:
+-  std::vector<prometheus::MetricFamily> Collect() const override;
++  std::vector<prometheus::MetricFamily> Collect() override;
```

Comment:
just curious, but why is it changed?

Reply:
Because prometheus-cpp mark it as const in the new version and this one update the code to make it compile. Don't want to upgrade prometheus-cpp since the changes are too big and seems not a blocker for std-20